File: examples/lh-server/typescript/src/09-external-events/main.ts
import { createTaskWorker, LHConfig, Workflow } from "littlehorse-client";
import { z } from "zod";
import { closeOnShutdown } from "../config.js";

function greet(name: string): string {
  const result = `Hello, ${name}!`;
  console.log(result);
  return result;
}

const config = LHConfig.from({});
const client = config.getClient();
const worker = createTaskWorker(greet, "greet", config, {
  inputVars: { name: z.string() },
  outputSchema: z.string(),
});

await client.putExternalEventDef({ name: "name-posted", contentType: {} });

const workflow = Workflow.newWorkflow("greet-event", (wf) => {
  const name = wf.declareStr("name");
  name.assign(wf.waitForEvent("name-posted"));
  wf.execute("greet", name);
});

await closeOnShutdown([worker]);
await worker.registerTaskDef();
await workflow.registerWfSpec(config);
console.log("Run with: lhctl run greet-event");
console.log("Then post with: npm run external-events:post -- <wf-run-id>");
await worker.start();

File: examples/lh-server/python/09-external-events/main.py
import asyncio
import sys

import littlehorse
from littlehorse.config import LHConfig
from littlehorse.worker import LHTaskWorker
from tasks import greet
from workflow import get_workflow


def register(config: LHConfig) -> None:
    littlehorse.create_task_def(greet, "greet", config)
    littlehorse.create_workflow_spec(get_workflow(), config)


async def workers(config: LHConfig) -> None:
    await littlehorse.start(LHTaskWorker(greet, "greet", config))


if __name__ == "__main__":
    config = LHConfig()
    if len(sys.argv) != 2 or sys.argv[1] not in {"register", "workers"}:
        raise SystemExit("Usage: main.py [register|workers]")
    register(config) if sys.argv[1] == "register" else asyncio.run(workers(config))
